src/app/src/actions/logDownload.js

```diff
@@ -0,0 +1,32 @@
+import { createAction } from 'redux-act';
+
+import apiRequest from '../util/apiRequest';
+
+import { makeLogDownloadUrl, logErrorAndDispatchFailure } from '../util/util';
+
+export const startLogDownload =
+    createAction('START_LOG_DOWNLOAD');
+export const failLogDownload =
+    createAction('FAIL_LOG_DOWNLOAD');
+export const completeLogDownload =
+    createAction('COMPLETE_LOG_DOWNLOAD');
+
+export function logDownload(rows, callback) {
+    return (dispatch) => {
+        dispatch(startLogDownload());
+
+        const path = `${window.location.pathname}${window.location.search}${window.location.hash}`;
+        const recordCount = rows ? rows.length : 0;
+        return apiRequest
+            .post(makeLogDownloadUrl(path, recordCount))
+            .then(() => {
+                callback(rows);
+                return dispatch(completeLogDownload());
+            })
```

Comment:
Nitpick but consider

```
.then(() => callback(rows))
.then(() => dispatch(completeLogDownload())
.catch....
```

Or whatever similar construction would work with a potential revision to use getState.

Reply:
👍

Reply:
Implemented in fixup c53103b